Library/Homebrew/command.rb

```diff
@@ -72,13 +72,51 @@ def self.desc(desc)
       end
     end
 
+    def self.get_invalid_options(argv_options_only)
+      argv_options_only = argv_options_only.uniq
+      valid_options_with_values =
+        @valid_options
+        .select { |option_hash| !option_hash[:value].nil? }
+        .map { |option_hash| option_hash[:option] }
+      all_valid_options =
+        @valid_options
+        .map { |option_hash| option_hash[:option] }
+
+      argv_invalid_options = []
+      argv_options_without_value = {}
+      argv_options_only.each do |argv_option|
```

Comment:
I think this entire loop could be flattened in a few places, to improve readability:

* Use `reject` to get the invalid options out, rather than handling them as a case in a condition.
* Use `select` to get the valid options with missing values out.